### PR TITLE
Trunk 4667

### PR DESCRIPTION
**Pull request by tendomart**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

https://issues.openmrs.org/browse/TRUNK-4667

@dkayiwa  @ibacher  i want to know wether am on the right track with the Newly created JPA repository and it's implementation , so that i can continue with the rest of the methods and the Unit Tests.
Don't mind about the Integration tests , i already run them locally.

### CODE REVIEW

File: api/src/main/resources/hibernate.default.properties [hunk 1]
```diff
@@ -5,6 +5,11 @@ hibernate.connection.username=test
 hibernate.connection.password=test
 hibernate.connection.driver_class=com.mysql.cj.jdbc.Driver
 hibernate.connection.url=jdbc:mysql://localhost:3306/openmrs?autoReconnect=true
+hibernate.cache.use_query_cache=false
+hibernate.hbm2ddl.auto=create-drop
+hibernate.ejb.naming_strategy=org.hibernate.cfg.ImprovedNamingStrategy
+
+
```
Comment (theanandankit) [inline api/src/main/resources/hibernate.default.properties:12]: Please remove the extra space.

File: api/src/main/java/org/openmrs/api/db/springdata/repository/JpaEncounterDao.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	public List<Encounter> getEncounters(EncounterSearchCriteria encounterSearchCriteria);
+
+	/**
+	 * Save an Encounter Type
+	 *
+	 * @param encounterType
+	 */
+	public EncounterType saveEncounterType(EncounterType encounterType);
+
+
```
Comment (theanandankit) [inline api/src/main/java/org/openmrs/api/db/springdata/repository/JpaEncounterDao.java:55]: Remove extra space

File: api/src/main/java/org/openmrs/api/db/springdata/SpringDataJpaConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+		for(Resource resource: resources){
+			entityManagerFactoryBean.setMappingResources(resource.getFile().getAbsolutePath());
+		}
+		Properties jpaProperties = new Properties();
+		jpaProperties.put("hibernate.ejb.interceptor", chainingInterceptor());
+		jpaProperties.load(new ClassPathResource("hibernate.default.properties").getInputStream());
+		jpaProperties.putAll(Context.getRuntimeProperties());
+		entityManagerFactoryBean.setJpaProperties(jpaProperties);
+
+
```
Comment (theanandankit) [inline api/src/main/java/org/openmrs/api/db/springdata/SpringDataJpaConfiguration.java:142]: Remove extra space

**Review by theanandankit — COMMENTED** [review]
@tendomart thanks, Please add the complete name of PR https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by tendomart** [comment]
Thanks @theanandankit

File: pom.xml [hunk 4]
```diff
@@ -596,6 +596,11 @@
 				<artifactId>jaxb-impl</artifactId>
 			    <version>3.0.0-M4</version>
 			</dependency>
+			<dependency>
+				<groupId>org.springframework.data</groupId>
```
Comment (sherrif10) [inline pom.xml:600]: Do you need this dependency

File: api/src/main/java/org/openmrs/BaseDosingInstructions.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+ */
+public abstract class BaseDosingInstructions implements DosingInstructions {
+
+	/**
+	 * @see DosingInstructions#getAutoExpireDate(DrugOrder)
+	 */
+	@Override
+	public Date getAutoExpireDate(DrugOrder drugOrder) {
+		if (drugOrder.getDuration() == null || drugOrder.getDurationUnits() == null) {
+			return null;
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/BaseDosingInstructions.java:27]: Isnt this throwing nullPointer Exceptions, i would have also added throws Exceptions with in this method

Comment (tendomart) [reply api/src/main/java/org/openmrs/BaseDosingInstructions.java:27]: Thanks for the check @sherrif10  , working on it ... Still a Draft though

File: pom.xml (hunk 4 above)
Comment (tendomart) [reply pom.xml:600]: Yes @sherrif10   Iam introducing  **spring-data-jpa** .
 Do you have a simpler way by which we can introduce **spring-data-jpa** ? or may be it's already available among the dependencies ?

File: api/src/main/java/org/openmrs/BaseDosingInstructions.java (hunk 5 above)
Comment (tendomart) [reply api/src/main/java/org/openmrs/BaseDosingInstructions.java:27]: Doing a close check it is actually still intact as it was originally.

**Pull request closed without merging**